plugin.sh.tmux

```diff
@@ -14,6 +14,34 @@ config_file="$root_dir/config.yaml"
 plugin_dir="$root_dir/plugin"
 init_file="$plugin_dir/init.tmux"
 
+# XDG
+case "$(tmux show-option -gvq @tmux-which-key-xdg-enable)" in
+    1 | true)
+        if [ -z "$XDG_CONFIG_HOME" ]; then
+            echo "[tmux-which-key] XDG_CONFIG_HOME is not set"
+            exit 1
+        fi
+        if [ ! -d "$XDG_CONFIG_HOME" ]; then
+            echo "[tmux-which-key] XDG_CONFIG_HOME: $XDG_CONFIG_HOME does not exist"
+            exit 1
+        fi
+        if [ -z "$XDG_DATA_HOME" ]; then
+            echo "[tmux-which-key] XDG_DATA_HOME is not set"
+            exit 1
+        fi
+        if [ ! -d "$XDG_DATA_HOME" ]; then
+            echo "[tmux-which-key] XDG_CONFIG_HOME: $XDG_DATA_HOME does not exist"
```

Comment:
Does the spec say to return an error if an XDG dir doesn't exist? I ask because it seems possible to just `mkdir -p` it if missing. No strong opinion either way

Reply:
I was being overly cautious, but I can make this change.

Yes, we can create the directory.  It does involve a little extra work.  I am thinking the basic flow is:

* If it exists, use it.
* If it doesn't, create it with 0700.
* Raise an error if for some reason these directories are not writeable or cannot be created.

Reply:
The spec states:

> The application should be prepared to handle the case where the file could not be written, either because the directory was non-existent and could not be created, or for any other reason. In such case it may choose to present an error message to the user.

With the earlier changes to use a default directory and create all paths that do not exist so long as they are contained in `$HOME`, this error would then be generated by `mkdir -p` in the new code.  In the existing code it would be generated by `cp` which is where it errors now with Nix.